src/duplicacy_snapshotmanager.go

```diff
@@ -1910,18 +1961,101 @@ func (manager *SnapshotManager) PruneSnapshots(selfID string, snapshotID string,
 		}
 	}
 
-	if toBeDeleted == 0 && exhaustive == false {
+	if toBeDeleted == 0 && !exhaustive {
 		LOG_INFO("SNAPSHOT_NONE", "No snapshot to delete")
 		return false
 	}
 
-	chunkRegex := regexp.MustCompile(`^[0-9a-f]+$`)
+	collection := CreateFossilCollection(allSnapshots)
 
-	referencedChunks := make(map[string]bool)
+	var success bool
```

Comment:
Thank you for splitting this into separate functions!  I have been too lazy and too afraid of breaking something.  Don't worry about the git diff results.  I'm sure a manual diff of the relevant parts should be good.

Reply:
No problem - it made particular sense to do it now, as the two paths now also differ in how they walk the snapshots.